NonBayesian/First/first.py

```diff
@@ -0,0 +1,153 @@
+import numpy as np
+from numpy.random import choice as generator
+import matplotlib.pyplot as plt
+
+def random_histogram(n):
+    res = np.random.rand(n)
+    return res / np.sum(res)
+
+
+# def generator(probability, length, k):
+# return np.random.choice(k, length, p=probability)
+
+
+def square_penalty(k, weight, j):
+    """
+        >>> [square_penalty([10, 20, 30, 40, 50],[0.1, 0.4, 0.2, 0.1, 0.2], i) for  i in range(0,5)]
+        [530.0, 250.0, 170.0, 290.0, 610.0]
+        """
+    res = 0
+    for i in range(0, len(k)):
+        res += weight[i] * (k[i] - k[j]) ** 2
+    return res
+
+
+def binary_penalty(k, weight, j):
+    """
+            >>> [binary_penalty([50, 13, 21, 48, 16],[0.7, 0.1, 0.1, 0.05, 0.05], i) for  i in range(0,5)]
+            [0.3, 0.9, 0.9, 0.95, 0.95]
+            """
+    res = 0
+    for i in range(0, len(k)):
+        res += weight[i] * (int(k[j] != k[i]))
+    return res
+
+
+def square_strategy(k, histogram):
+    """
+                >>> square_strategy([50, 13, 21, 48, 16, 79],[0.3, 0.1, 0.1, 0.05, 0.05, 0.4])
+                [0, 50]
+                """
+    res = 0
+    minimum = square_penalty(k, histogram, 0)
+    for j in range(1, len(k)):
+        a = square_penalty(k, histogram, j)
+        if a < minimum:
+            minimum = a
+            res = j
+    return [res, k[res]]
```

Comment:
It's prettier to return tuples in such cases
```suggestion
    return res, k[res]
```